src/main/java/app/studnicki/ox/BoardChecker.java

```diff
@@ -14,49 +11,41 @@
  *
  * @author Aleksander Studnicki
  */
-class BoardChecker implements PropertyChangeListener {
+class BoardChecker {
 
   private final int winningRule;
-  private final int checkingThreshold;
   private final Set<TransitionRule> ruleSet = EnumSet.of(
       VERTICAL, HORIZONTAL, DIAGONAL_DOWN, DIAGONAL_UP);
   private int turn;
+  private Board board;
+  private boolean ableToCheck = true;
 
-  private final PropertyChangeSupport propertyChangeSupport = new PropertyChangeSupport(this);
+  BoardChecker(int winningRule, Board board) {
+    this.winningRule = winningRule;
+    this.board = board;
+  }
 
   /**
-   * Adds observers for 2 type of events.
-   * Playable - when there is possibility to play and no winner.
-   * Resolved - when there is a winner or no possibility to play (draw).
-   * When resolved then flag is send to observer (true - winner, false - draw).
+   * Checks if there is a winner after last field marking.
    *
-   * @param listener Object which will be observing GameChecker
-   *                 (must implements PropertyChangeListener)
+   * @param id id of recent marked field
+   * @return true if is winner and false otherwise
    */
-  void addListener(PropertyChangeListener listener) {
-    propertyChangeSupport.addPropertyChangeListener("playable", listener);
-    propertyChangeSupport.addPropertyChangeListener("resolved", listener);
-  }
-
-  BoardChecker(int winningRule) {
-    this.winningRule = winningRule;
-    this.checkingThreshold = winningRule * 2 - 1;
+  boolean isWinner(int id) {
+    turn++;
+    return isWinner(id, board);
   }
 
-  boolean isWinner(int id, Board board) {
+  private boolean isWinner(int id, Board board) {
```

Comment:
Czy ta metoda, aby na pewno jest potrzebna? `isWinner(int id)` korzysta z niej i zwiększa `turn`, skoro `board` jest polem to mamy do niego dostęp. Moim zdaniem logikę tej metody wywaliłbym do tej package-private.

Reply:
Zmienione.